src/deprecated.jl

```diff
@@ -22,3 +22,6 @@ import Base: keys, values, insert!
 
 @deprecate stackdf stackdt
 @deprecate meltdf meltdt
+
+@deprecate readtable CSV.read
```

Comment:
I see here https://github.com/JuliaLang/julia/blob/master/base/deprecated.jl#L509 that deprecations can accept keyword arguments, so I take it this is a hint to do the same :).

`header=false -> datarow=1` 
`separator -> delim`
`quotemark -> quotechar`
`decimal -> ??`
`nastrings -> null`
`truestrings -> ??`
`falsestrings -> ??`
`makefactors -> ??`
`nrows -> rows`
`names -> header`
`eltypes -> types`
`allowcomments -> ??`
`commentmark -> ??`
`ignorepadding -> ??`
`skipstart -> datarow`
`skiprows -> ??`
`skipblanks -> ??`
`encoding -> ??`
`normalizenames -> ??`
`?? -> escapechar`
`?? -> nullable`
`?? -> weakrefstrings` this might be equivalent to make factors, need to check
`?? -> dateformat`
`?? -> footerskip`
`?? -> rows_for_type_detect`
`?? -> use_mmap`
`?? -> append`
`?? -> transforms`